src/sagemaker/local/utils.py

```diff
@@ -72,6 +72,22 @@ def move_to_destination(source, destination, job_name, sagemaker_session):
     return final_uri
 
 
+def _create_s3_prefix(path, job_name):
+    """Constructs a path out of the given path and job name to be
+    used as an S3 prefix.
+
+    Args:
+        path (str): the original path. If the path is only ``"/"``,
+            then it is ignored.
+        job_name (str): the job name to be appended to the path.
+
+    Returns:
+        str: an S3 prefix of the form ``"path/job_name"``
+    """
+    path = path.strip("/")
+    return job_name if path == "" else "/".join((path, job_name))
```

Comment:
Just curious to know instead of doing string manipulation, wouldn't `os.path.join` or `pathlib.Path` a better options for correctly making unified paths for local and non-local modes?

Reply:
it would, except that S3 always uses "/" as a separator, so using `os.path.join` would fail for Windows users.

as far as I can tell, there's not an easy way of overriding the separator used in `os.path.join` (I wrote a little about this in #1435). The other option would be to use something like [urlunparse](https://docs.python.org/3.7/library/urllib.parse.html#urllib.parse.urlunparse), but I think it might be a little heavy-handed for this use-case.

Reply:
`PurePosixPath` does actually look to solve the problem, except that it was introduced in Python 3.4 and never backported to Python 2.7. since this SDK still currently supports Python 2.7, we can't use it here either. however, I'll make a note in our internal backlog to use `PurePosixPath` after we do get around to dropping Python 2.7 support. Thanks for sharing that! We've had quite a few conversations on our team around finding a module that would work no matter the OS, but hadn't yet found `pathlib`.

Reply:
Sounds great! thanks for letting me know.